Fix false positive warnings with global CFNotificationCenters

fred-wang opened the pull request:
#### 16ffde07240e6edc23df072d113cb0a60f43c883
<pre>
Fix false positive warnings with global CFNotificationCenters
<a href="https://bugs.webkit.org/show_bug.cgi?id=295778">https://bugs.webkit.org/show_bug.cgi?id=295778</a>

Reviewed by Darin Adler and Ryosuke Niwa.

Introduce &quot;*Singleton&quot; versions of the following functions to turn off clang safer cpp warnings:

CFNotificationCenterGetDarwinNotifyCenter()
CFNotificationCenterGetDistributedCenter()
CFNotificationCenterGetLocalCenter()

* Source/WTF/WTF.xcodeproj/project.pbxproj: Add header.
* Source/WTF/wtf/CMakeLists.txt: Ditto.
* Source/WTF/wtf/PlatformMac.cmake: Ditto.
* Source/WTF/wtf/cf/LanguageCF.cpp:
(WTF::listenForLanguageChangeNotifications): Use singleton version.
* Source/WTF/wtf/cf/NotificationCenterCF.h: Added.
(WTF::CFNotificationCenterGetDarwinNotifyCenterSingleton): Define singleton version.
(WTF::CFNotificationCenterGetLocalCenterSingleton): Ditto.
(WTF::CFNotificationCenterGetDistributedCenterSingleton): Ditto.
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp: Use singleton version.
(WebCore::CaptionUserPreferencesMediaAF::~CaptionUserPreferencesMediaAF): Ditto.
(WebCore::CaptionUserPreferencesMediaAF::setInterestedInCaptionPreferenceChanges): Ditto.
* Source/WebCore/platform/cf/MainThreadSharedTimerCF.cpp:
(WebCore::setupPowerObserver): Ditto.
* Source/WebCore/platform/cocoa/EffectiveRateChangedListener.mm:
(WebCore::EffectiveRateChangedListener::EffectiveRateChangedListener): Ditto.
(WebCore::EffectiveRateChangedListener::stop): Ditto.
* Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
(WebCore::registerNotificationForWebContentFilterTypeChange): Ditto.
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::FontCache::platformInit): Ditto.
* Source/WebCore/platform/text/cocoa/LocalizedDateCache.mm:
(WebCore::LocalizedDateCache::LocalizedDateCache): Ditto.
(WebCore::LocalizedDateCache::~LocalizedDateCache): Ditto.
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations: Remove files with warnings fixed.
* Source/WebKit/UIProcess/API/Cocoa/_WKSystemPreferences.mm:
(+[_WKSystemPreferences setCaptivePortalModeEnabled:]): Ditto.
(+[_WKSystemPreferences setCaptivePortalModeIgnored:ignore:]): Ditto.
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::registerNotificationObservers): Ditto.
(WebKit::WebProcessPool::unregisterNotificationObservers): Ditto.
(WebKit::WebProcessPool::registerHighDynamicRangeChangeCallback): Ditto.
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/ios/WKWebGeolocationPolicyDeciderIOS.mm:
(-[WKWebGeolocationPolicyDecider init]): Ditto.
(-[WKWebGeolocationPolicyDecider dealloc]): Ditto.
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::colorPreferencesDidChange): Ditto.
(WebKit::WebProcess::dispatchSimulatedNotificationsForPreferenceChange): Ditto.
* Source/WTF/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations: Remove files with warning fixed.

Canonical link: <a href="https://commits.webkit.org/299746@main">https://commits.webkit.org/299746@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/89187bc2068feff3079c1f154fa7c9c4a6bc5ecb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120042 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39736 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30387 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126384 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72115 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121918 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40432 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48313 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91162 "23 flakes 74 failures") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60471 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122994 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32295 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107636 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71716 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31330 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25740 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70012 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/112167 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101774 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25928 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129292 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/118558 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46962 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35616 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99777 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47328 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103823 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99621 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25297 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45077 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23099 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43585 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46824 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52530 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/147257 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46290 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37839 "Found 1 new JSC binary failure: testapi, Found 18678 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/nativearray_gen6.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse2.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/blue_1086262.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/bug215238.shrua-2.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default ... (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49639 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47976 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WTF/wtf/cf/NotificationCenterCF.h:
```diff
@@ -0,0 +1,49 @@
+    return CFNotificationCenterGetLocalCenter();
+}
+
+} // namespace WTF
```
Can we add `using` here so that we don't need to explicitly qualify with WTF?

fred-wang committed:
Fix false positive warnings with global CFNotificationCenters
https://bugs.webkit.org/show_bug.cgi?id=295778

Reviewed by Darin Adler and Ryosuke Niwa.

Introduce "*Singleton" versions of the following functions to turn off clang safer cpp warnings:

CFNotificationCenterGetDarwinNotifyCenter()
CFNotificationCenterGetDistributedCenter()
CFNotificationCenterGetLocalCenter()

* Source/WTF/WTF.xcodeproj/project.pbxproj: Add header.
* Source/WTF/wtf/CMakeLists.txt: Ditto.
* Source/WTF/wtf/PlatformMac.cmake: Ditto.
* Source/WTF/wtf/cf/LanguageCF.cpp:
(WTF::listenForLanguageChangeNotifications): Use singleton version.
* Source/WTF/wtf/cf/NotificationCenterCF.h: Added.
(WTF::CFNotificationCenterGetDarwinNotifyCenterSingleton): Define singleton version.
(WTF::CFNotificationCenterGetLocalCenterSingleton): Ditto.
(WTF::CFNotificationCenterGetDistributedCenterSingleton): Ditto.
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp: Use singleton version.
(WebCore::CaptionUserPreferencesMediaAF::~CaptionUserPreferencesMediaAF): Ditto.
(WebCore::CaptionUserPreferencesMediaAF::setInterestedInCaptionPreferenceChanges): Ditto.
* Source/WebCore/platform/cf/MainThreadSharedTimerCF.cpp:
(WebCore::setupPowerObserver): Ditto.
* Source/WebCore/platform/cocoa/EffectiveRateChangedListener.mm:
(WebCore::EffectiveRateChangedListener::EffectiveRateChangedListener): Ditto.
(WebCore::EffectiveRateChangedListener::stop): Ditto.
* Source/WebCore/platform/cocoa/ParentalControlsURLFilter.mm:
(WebCore::registerNotificationForWebContentFilterTypeChange): Ditto.
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::FontCache::platformInit): Ditto.
* Source/WebCore/platform/text/cocoa/LocalizedDateCache.mm:
(WebCore::LocalizedDateCache::LocalizedDateCache): Ditto.
(WebCore::LocalizedDateCache::~LocalizedDateCache): Ditto.
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations: Remove files with warnings fixed.
* Source/WebKit/UIProcess/API/Cocoa/_WKSystemPreferences.mm:
(+[_WKSystemPreferences setCaptivePortalModeEnabled:]): Ditto.
(+[_WKSystemPreferences setCaptivePortalModeIgnored:ignore:]): Ditto.
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::registerNotificationObservers): Ditto.
(WebKit::WebProcessPool::unregisterNotificationObservers): Ditto.
(WebKit::WebProcessPool::registerHighDynamicRangeChangeCallback): Ditto.
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/ios/WKWebGeolocationPolicyDeciderIOS.mm:
(-[WKWebGeolocationPolicyDecider init]): Ditto.
(-[WKWebGeolocationPolicyDecider dealloc]): Ditto.
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::colorPreferencesDidChange): Ditto.
(WebKit::WebProcess::dispatchSimulatedNotificationsForPreferenceChange): Ditto.
* Source/WTF/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations: Remove files with warning fixed.

Canonical link: https://commits.webkit.org/299746@main
